src/Corax/Querying/IndexSearcher.cs

```diff
@@ -325,25 +327,24 @@ private ByteStringContext<ByteStringMemoryCache>.InternalScope AnalyzeTerm(Analy
 
    public long GetDictionaryIdFor(Slice field)
    {
-       var terms = _fieldsTree?.CompactTreeFor(field);
-       return terms?.DictionaryId ?? -1;
+       if (_fieldsTree == null || _fieldsTree.TryGetCompactTreeFor(field, out var terms) == false)
+            return -1;
+
+       return terms.DictionaryId;
    }
    
     public long GetTermAmountInField(in FieldMetadata field)
     {
-        long termAmount = 0;
-        
-        var fieldTree = _fieldsTree?.CompactTreeFor(field.FieldName);
+        long termAmount = 0;  
+        if (_fieldsTree != null && _fieldsTree.TryGetCompactTreeFor(field.FieldName, out var fieldTree))
+            termAmount += fieldTree.NumberOfEntries;
 
-        termAmount += fieldTree?.NumberOfEntries ?? 0;
-
-        if (TryGetPostingListForNull(field, out var nullPostingListId))
-        {
-            var nullPostingList = GetPostingList(nullPostingListId);
-
-            termAmount += nullPostingList?.State.NumberOfEntries ?? 0;
-        }
+        if (!TryGetPostingListForNull(field, out var nullPostingListId)) 
```

Comment:
Can we use `== false` instead of `!`? Just for readability purposes

Reply:
Yes, I missed that one.

Reply:
I saw it in other places as well